service/history/engine/engineimpl/historyEngine.go

```diff
@@ -2584,6 +2584,10 @@ func (e *historyEngineImpl) SignalWithStartWorkflowExecution(
 				return nil, workflow.ErrSignalsLimitExceeded
 			}
 
+			if requestID := sRequest.GetRequestID(); requestID != "" {
```

Comment:
I mean, pretty amazingly big fix, nice find. Is this flow covered with a unit-test? If they're not, I think we ought to probably cover all write (+query) APIs to ensure they're idempotent